snuba/web/split.py

```diff
@@ -128,61 +174,100 @@ def time_split(dataset, request: Request, *args, **kwargs):
 
         return overall_result
 
-    def col_split(
-        dataset, request: Request, column_split_spec: ColumnSplitSpec, *args, **kwargs
-    ):
+
+class ColumnSplitQueryStrategy(StorageQuerySplitStrategy):
+    """
+    A strategy that performs column based splitting: if the client requests enough columns,
+    a first query on the minimum set of columns is ran to load as little Clickhouse data
+    as possible. A second query based on the results of the first is then executed to
+    build the full result set.
+    """
+
+    def __init__(
+        self, id_column: str, project_column: str, timestamp_column: str
+    ) -> None:
+        self.__id_column = id_column
+        self.__project_column = project_column
+        self.__timestamp_column = timestamp_column
+
+    def __get_min_columns(self) -> Sequence[str]:
+        return [self.__id_column, self.__project_column, self.__timestamp_column]
+
+    def __can_execute(self, request: Request) -> bool:
+        (use_split,) = state.get_configs([("use_split", 0)])
+        if not use_split:
+            return False
+
+        if not _is_query_splittable(request.query):
+            return False
+
+        if not request.query.get_selected_columns():
+            return False
+
+        if request.query.get_aggregations():
+            return False
+
+        total_col_count = len(request.query.get_all_referenced_columns())
+        copied_query = copy.deepcopy(request.query)
```

Comment:
This copies twice now. 😞

Reply:
The next statement uses the copy and modifies it to decide whether we can split.
I can move this into the execute method instead. But the copy is needed to decide whether the splitter can be executed.

Reply:
The end result of the copy and `set_selected_columns` is the same as line 232-233, isnt it?

The double copy issue was also mentioned here: https://github.com/getsentry/snuba/pull/865#discussion_r404557527

Reply:
Well, I think it is better to just remove the can_execute methods